### PR TITLE
get single currency from api

### PR DESCRIPTION
**Pull request by I-Behdar**

### CODE REVIEW

File: currency_exchange/app.py [hunk 1]
```diff
@@ -2,47 +2,42 @@
...
-    for i in rates:
-        if i["code"] == currency:
-            return i["mid"]
 
 def convert(value: float, currency: str) -> float:
     exchange_rate = fetch_currency_rate(currency)
-    return value * exchange_rate
-    # return exchange_rate * value
-
+    return round(value * exchange_rate, 2)
```
Comment (mstopa) [inline currency_exchange/app.py]: Please add `if __name__ == '__main__'` in the initial point of the program flow - https://realpython.com/if-name-main-python/

File: currency_exchange/app.py [hunk 2]
```diff
@@ -2,47 +2,42 @@
...
 
 
 input = (args.amount, args.currency)
```
Comment (mstopa) [inline currency_exchange/app.py]: Please move `args = parser()` from line 25 to the closest line up to `input = (args.amount, args.currency)` (currently line 40)

File: currency_exchange/app.py [hunk 3]
```diff
@@ -2,47 +2,42 @@
...
 output = convert(value=input[0], currency=input[1])
```
Comment (mstopa) [inline currency_exchange/app.py]: We no longer need `input` tuple, please use `args.amount`, `args.currency` directly in the function call

File: currency_exchange/app.py [hunk 4]
```diff
@@ -2,47 +2,42 @@
 import argparse
 import requests
 
-
 CURRENCIES = {
```
Comment (mstopa) [inline currency_exchange/app.py]: we no longer need this dict.

After you remove it, we can no longer limit choices in args.currency as we used to:
```
    parser.add_argument('currency', type=str,
                        choices=CURRENCIES.keys(), help='currency type')
```

Please read https://realpython.com/python-exceptions/ and handle the case when a user requests a currency that doesn't exist so `r = requests.get(f'http://api.nbp.pl/api/exchangerates/rates/A/{currency}/')` fails

**Review by mstopa — CHANGES_REQUESTED** [review]

File: currency_exchange/app.py (hunk 1 above)
Comment (I-Behdar) [reply currency_exchange/app.py]: Done

File: currency_exchange/app.py (hunk 2 above)
Comment (I-Behdar) [reply currency_exchange/app.py]: Done

File: currency_exchange/app.py (hunk 3 above)
Comment (I-Behdar) [reply currency_exchange/app.py]: Done

File: currency_exchange/app.py (hunk 4 above)
Comment (I-Behdar) [reply currency_exchange/app.py]: Done

**Comment by mstopa** [comment]
@I-Behdar please only update requirements.txt and setup.py which currently contain some mock example data, after that it LGTM so you can can click on `Merge pull request` and merge your develop branch to main

**Comment by I-Behdar** [comment]
Done

**Pull request merged**